CONTRIBUTING.md

```diff
@@ -0,0 +1,67 @@
+# Contribute to Doctrine
+
+Thank you for contributing to Doctrine!
+
+Before we can merge your Pull-Request here are some guidelines that you need to follow.
+These guidelines exist not to annoy you, but to keep the code base clean,
+unified and future proof.
+
+## We only accept PRs  to "master"
+
+Our branching strategy is "everything to master first", even
+bugfixes and we then merge them into the stable branches. You should only 
+open pull requests against the master branch. Otherwise we cannot accept the PR.
+
+There is one exception to the rule, when we merged a bug into some stable branches
+we do occasionally accept pull requests that merge the same bug fix into earlier
+branches.
+
+## Coding Standard
+
+We use [doctrine coding standard](https://github.com/doctrine/coding-standard) which is PSR-1 and PSR-2:
+
+* https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-1-basic-coding-standard.md
+* https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md
+
+with some exceptions/differences:
+
+* Keep the nesting of control structures per method as small as possible
+* Align equals (=) signs
+* Add spaces between assignment, control and return statements
+* Prefer early exit over nesting conditions
+* Add spaces around a negation if condition ``if ( ! $cond)``
+* Add legal information at the beginning of each source file
+* Add ``@author`` [phpDoc](https://www.phpdoc.org/docs/latest/references/phpdoc/tags/author.html) comment at DockBlock of class/interface/trait that you create.
+
+## Unit-Tests
+
+Please try to add a test for your pull-request.
+
+* If you want to contribute new functionality add unit- or functional tests
+  depending on the scope of the feature.
+
+You can run the unit-tests by calling ``vendor/bin/phpunit`` from the root of the project.
+It will run all the project tests.
+
+In order to do that, you will need a fresh copy of doctrine/collections, and you
+will have to run a composer installation in the project:
+
+```sh
+git clone git@github.com:doctrine/collections.git
+cd collections
+curl -sS https://getcomposer.org/installer | php --
+./composer.phar install
+```
+
+## Travis
+
+We automatically run your pull request through [Travis CI](http://www.travis-ci.org)
+against PHP 5.5, PHP 5.6, PHP 7 and HHVM. If you break the tests, we cannot merge your code,
```

Comment:
I would say ``against supported PHP versions`` to avoid having to maintain this list when it changes